# Overview

![](.gitbook/assets/cellxgene_logo.svg)

Cellxgene is a collection of computational tools enabling the exploration, analysis, and reuse of single cell data. The [cellxgene data portal](https://cellxgene.cziscience.com/) is a place where you can search and download publicly available single cell datasets. Each dataset that is hosted on the cellxgene data portal is accompanied by a hosted instance of the cellxgene explorer, enabling frictionless visualization, exploration, and basic analysis of single cell data. To serve use cases where it is not possible to submit data to our portal, cellxgene is also available as a "desktop" explorer which can be used locally or in a self hosted setup. Here is a brief rundown of the features in cellxgene explorer, along with the notable differences between the desktop and hosted explorers:

* Coloring by categorial meta \(i.e. cell type, batch, treatment group, donor origin...\) and continuous metadata \(i.e. gene expression, qc metadata, prediction scores...\)
* Value clipping when coloring by continuous metadata \(when outliers distort the color scale\)
* Cell selection based on categorical metadata, numeric selection based on continuous metadata via range selection, or manual selection via lasso tool.
* Breakdown of specific categorical metadata fields \(i.e. tissue\) by any other categorical metadata field \(i.e. cell type\)
* * Embedding display and support for multiple embeddings \(umap, tSNE, pva, pseudotime...\)
* Recalculate embeddings: Desktop Explorer Only
* Custom cell type annotations: Desktop Explorer Only

Table of features \(desktop explorer vs hosted explorer\)

Link out to appropriate docs sections \(quick re-direction to get people to places we think will be most commonly visited\)

link to cellxgene data portal, single-cell-curation-repo, schema documents

Link to cellxgene github repo \(already represented in the code section, not needed?\)







